### PR TITLE
Hyperband Refector

### PR DESCRIPTION
**Pull request by J-shang**
1. fix a bug that may cover unfinished bracket.
2. add 'serial' mode and named the original mode 'parallelism'.
3. add a test for hyperband using nasbench201.
4. Modify the document.

### CODE REVIEW

**Comment by QuanluZhang** [comment]
@J-shang looks good. it might be better to add the result figure for nasbench201 in the doc

**Review by colorjam — APPROVED** [review]
Looks great~

File: nni/algorithms/hpo/hyperband_advisor/test/main.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+
+logger = logging.getLogger('test_hyperband')
+
+
+def main(args):
+    r = args.pop('TRIAL_BUDGET')
+    dataset = [t for t in query_nb201_trial_stats(args, 200, 'cifar100', include_intermediates=True)]
+    test_acc = random.choice(dataset)['intermediates'][r - 1]['ori_test_acc'] / 100
+
```
Comment (QuanluZhang) [inline nni/algorithms/hpo/hyperband_advisor/test/main.py:23]: better to add some random delay here to further check the correctness of the implementation

File: nni/algorithms/hpo/hyperband_advisor/test/config_hyperband.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+authorName: default
```
Comment (QuanluZhang) [inline nni/algorithms/hpo/hyperband_advisor/test/config_hyperband.yml:1]: @liuzhe-lz is this a correct place to put tests?

File: nni/algorithms/hpo/hyperband_advisor/test/main.py (hunk 1 above)
Comment (J-shang) [reply nni/algorithms/hpo/hyperband_advisor/test/main.py:23]: Okey~

File: nni/algorithms/hpo/hyperband_advisor/test/config_hyperband.yml (hunk 2 above)
Comment (J-shang) [reply nni/algorithms/hpo/hyperband_advisor/test/config_hyperband.yml:1]: I will check it if this can move to UT.

Comment (J-shang) [reply nni/algorithms/hpo/hyperband_advisor/test/config_hyperband.yml:1]: This test need nasbench201, and installing nasbench201 will cost 2h in my machine, so maybe temporarily put it here?

**Review by QuanluZhang — APPROVED** [review]

**Review by liuzhe-lz — APPROVED** [review]

**Pull request merged**